tests/PHPStan/Analyser/nsrt/bug-5017.php

```diff
@@ -15,7 +15,7 @@ public function doFoo()
 			assertType('non-empty-array<0|1|2|3|4, 0|1|2|3|4>', $items);
 			$batch = array_splice($items, 0, 2);
 			assertType('array<0|1|2|3|4, 0|1|2|3|4>', $items);
-			assertType('non-empty-list<0|1|2|3|4>', $batch);
+			assertType('list<0|1|2|3|4>', $batch);
```

Comment:
this is unfortunate, but it happens because the `$items` type is not as good as it can be. This will be improved in https://github.com/phpstan/phpstan-src/pull/3952

So there's unfortunately some kind of coupling here.